### ISSUE TITLE
Records containing arrays pretty print poorly

### ISSUE THREAD

**Issue by oxinabox** [google-research/dex-lang#320]
Consider:
```
>=>  :p {a=1, b=2}
{a = 1, b = 2}
```

vs
```
>=>:p {a="12", b=[1,2,3]}
{ a = \for i:(Fin 2) =>
  tmp:Int32 = %toOrdinal i
  tmp1:(Ptr (d, CPU, Int8)) = Ptr (a, CPU, Int8) 0x00007fb77161c8c0 +> tmp
  tmp2:Int8 = load tmp1
  Char tmp2
, b = \for i:(Fin 3) =>
  tmp:Int32 = %toOrdinal i
  tmp1:(Ptr (d, CPU, Int32)) = Ptr (a, CPU, Int32) 0x00007fb771616e40 +> tmp
  load tmp1}
```

**Comment by dan-zheng** [comment]
I'll look into this. It is related to the `prettyVal` function, which is missing a special-case for `Record`.

https://github.com/google-research/dex-lang/blob/55d9ae9309363bdc386bbc7c8654c9e3cd738953/src/lib/Serialize.hs#L39-L40

**Comment by apaszke** [comment]
Fixed by #327!
